scripts/mobile_generator.py

```diff
@@ -163,4 +128,151 @@ def __lt__(self, other):
+            result += f'        let {member.name}: Int'
+        elif member.type == 'boolean':
+            result += f'        let {member.name}: Bool'
+        else:
+            raise Exception(f"Not handled yet: {member.type}")
+        result += f"{defaultValue}\n"
+
+    for enum in enums:
+        result += "\n"
+        result += f"        enum {enum.name}: String {{\n"
+        enum.values.sort()
+        for value in enum.values:
+            result += f"            case {value}\n"
+        result += "        }\n"
+
+    result += "\n"
+    if not members:
+        result += "        var properties: [String: Any] = [:]\n"
+    else:
+        filteredMembers = list(filter(lambda member: member.name != "screenName", members))
+        result += "        var properties: [String: Any] {\n"
+        result += "            return [\n"
+        for index, member in enumerate(filteredMembers):
+            if member.enum:
+                if member.required:
+                    result += f'                "{member.name}": {member.name}.rawValue'
+                else:
+                    result += f'                "{member.name}": {member.name}?.rawValue as Any'
+            else:
+                if member.required:
+                    result += f'                "{member.name}": {member.name}'
+                else:
+                    result += f'                "{member.name}": {member.name} as Any'
+            if index < len(filteredMembers) - 1:
+                result += ",\n"
+            else:
+                result += "\n"
+        result += "            ]\n"
+        result += "        }\n"
+
+    result += "    }\n}"
+    return result
+
+@dataclass
+class Enum():
+    name: str
+    values: list[object]
+        
+@dataclass
+class Member():
+    name:str
+    type: str
+    enum: list[str]
+    description: str
+    required: bool
+    def __lt__(self, other):
+        return self.name < other.name
+        
+with open(args.json_schema) as json_file:
+    klass = os.path.basename(args.json_schema).removesuffix(".json")
+    isScreen = klass == "Screen"
+    data = json.load(json_file)
+
+    # Parse
+    members = []
+    enums = []
+    name = data['properties']['eventName']['enum'][0]
+    required = data.get('required')
+    for p in data['properties']:
+        if p == 'eventName':
+            continue
+        enum = data['properties'][p].get('enum')
+        if enum:
+            enums.append(Enum(first_letter_up(p), enum))
+
+        members.append(
+            Member(
+                p,
+                data['properties'][p].get('type'),
+                enum,
+                data['properties'][p].get('description'),
+                p in required or data['properties'][p].get('required')
+                )
+            )
+    members.sort()
+    
+    # Compute output
+    if args.output_language == "swift":
+        print(compute_swift())
+    else:
+        print(compute_kotlin())
```

Comment:
Those two functions should take `members` and `enums` as parameters, here those two lists are global and it makes me a bit uncomfortable.

Reply:
Ah yes, and `name`. All added.

Reply:
Thanks. Maybe we can have a first fun which parse the data, for better separation of concerns.

Reply:
Good plan 👍

Reply:
Updated